src/combinatorics/simple_graph/inc_matrix.lean

```diff
@@ -184,4 +197,170 @@ begin
 end
 
 end semiring
+
+/-! ## Orientations -/
+
+section orientations
+
+/-- Define an `orientation` on the simple graph `G` as a structure that defines (consistently)
+for each edge a `head` and a `tail`. -/
+@[ext]
+structure orientation (G : simple_graph α) :=
+(head : G.edge_set → α)
+(tail : G.edge_set → α)
+(consistent (e : G.edge_set) : ↑e = ⟦(head e, tail e)⟧)
+
+noncomputable instance (G : simple_graph α) : inhabited (orientation G) :=
+⟨{ head := λ (e : G.edge_set), (quotient.out (e : sym2 α)).fst,
+  tail := λ (e : G.edge_set), (quotient.out (e : sym2 α)).snd,
+  consistent := λ (e : G.edge_set), by rw [prod.mk.eta, quotient.out_eq] }⟩
+
+variables {o : orientation G}
+
+lemma head_tail (o : orientation G) (i : α) (e : G.edge_set) :
+  i = o.head e ∨ i = o.tail e ∨ (i ≠ o.head e ∧ i ≠ o.tail e) :=
+by tauto
```

Comment:
This lemma seems odd, isn't this just a general statement about equality of three items?